designer/client/src/PageEdit.tsx

```diff
@@ -141,22 +142,24 @@ export class PageEdit extends Component<Props, State> {
     const { save, data } = this.context
     const { page, onSave } = this.props
 
-    const copy = clone(data)
-    const copyPageIdx = copy.pages.findIndex((p) => p.path === page.path)
+    let copy = structuredClone(data)
+    const { pages } = copy
+
+    const pageRemove = findPage(copy, page.path)
+    const pageIndex = pages.indexOf(pageRemove)
 
     // Remove all links to the page
-    copy.pages.forEach((p, index) => {
-      if (index !== copyPageIdx && Array.isArray(p.next)) {
-        for (let i = p.next.length - 1; i >= 0; i--) {
-          const next = p.next[i]
-          if (next.path === page.path) {
-            p.next.splice(i, 1)
-          }
-        }
+    for (const pageFrom of pages.filter(hasNext)) {
```

Comment:
Just to avoid confusion, I'm going to change `pages` → `copy.pages`

Important to remember that `deleteLink` clones `copy` so `{ pages }` is from another reference

```suggestion
    for (const pageFrom of copy.pages.filter(hasNext)) {
```